xbmc/pvr/recordings/PVRRecording.h

```diff
@@ -82,8 +84,23 @@ namespace PVR
     std::string   m_strFanartPath;    /*!< fanart path */
     unsigned      m_iRecordingId;     /*!< id that won't change while xbmc is running */
 
+  private:
     CPVRRecording(void);
     CPVRRecording(const PVR_RECORDING &recording, unsigned int iClientId);
+
+    // Prevent copy construction, even for CPVRRecording instances and friends.
+    // Note: Only declared, but intentionally not implemented
+    //       to prevent compiler generated copy ctor and to force
+    //       a linker error in case somebody tries to call it.
+    CPVRRecording(const CPVRRecording &tag);
```

Comment:
BTW: These member functions are already declared in the private part. And only delcaring but not implementing them prevents not just friends from copying, but even instances of CPVRRecording itself. Maximum safety wrt. early error prevention, but no drawbacks here. Don't see what implementing these member functions has on the plus side.